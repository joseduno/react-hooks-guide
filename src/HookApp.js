import React from 'react'
import MainApp from './components/09-useContext/MainApp'
import TodoApp from './components/08-useReducer/TodoApp'
import { Padre } from './components/07-tarea-memo/Padre'
import CallbackHook from './components/06-memos/CallbackHook'
import MemoHook from './components/06-memos/MemoHook'
import Memorize from './components/06-memos/Memorize'
import Layout from './components/05-useLayoutEffect/Layout'
import RealExampleRef from './components/04-useRef/RealExampleRef'
import FocusScreen from './components/04-useRef/FocusScreen'
import MultipleCustomHooks from './components/03-examples/MultipleCustomHooks'
import FormWithCustomHook from './components/02-useEffect/FormWithCustomHook'
import CounterApp from './components/01-useState/CounterApp'
import CustomCounterHook from './components/01-useState/CustomCounterHook'
import SimpleForm from './components/02-useEffect/SimpleForm'

const HookApp = () => {
    return (
        <div>
            <CounterApp />
            <CustomCounterHook />
            <SimpleForm />
            <FormWithCustomHook />
            <MultipleCustomHooks />
            <FocusScreen />
            <RealExampleRef />
            <Layout />
            <Memorize />
            <MemoHook />
            <CallbackHook />
            <Padre />
            <TodoApp />
            <MainApp />
            {/* <h1>Hola Mundo!</h1> */}
        </div>
    )
}

export default HookApp
