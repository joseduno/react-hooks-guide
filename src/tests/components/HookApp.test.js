import { shallow } from 'enzyme'
import HookApp from '../../HookApp'

describe('Pruebas sobre el componente Hook App', () => {
    test('debe renderizar correctamente el componentes', () => {
        const wrapper = shallow(<HookApp />)
        expect(wrapper).toMatchSnapshot()
    })
    
    
})
