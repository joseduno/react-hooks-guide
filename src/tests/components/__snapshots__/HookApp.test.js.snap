// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente Hook App debe renderizar correctamente el componentes 1`] = `
<div>
  <CounterApp />
  <CustomCounterHook />
  <SimpleForm />
  <FormWithCustomHook />
  <MultipleCustomHooks />
  <FocusScreen />
  <RealExampleRef />
  <Layout />
  <Memorize />
  <MemoHook />
  <CallbackHook />
  <Padre />
  <TodoApp />
  <MainApp />
</div>
`;
